vivarium/core/experiment.py

```diff
@@ -850,24 +888,20 @@ def state_for(self, path, keys):
                 key: state.inner_value(key)
                 for key in keys}
 
-    def depth(self, path=()):
+    def depth(self, path=(), predicate=None):
```

Comment:
I'm not sure what `predicate` means here. Maybe `filter`?

Reply:
thanks, `filter` is better.